packer/builder/azure/arm/step_delete_os_disk.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See the LICENSE file in the project root for license information.
+
+package arm
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/Azure/packer-azure/packer/builder/azure/common/constants"
+
+	"github.com/mitchellh/multistep"
+	"github.com/mitchellh/packer/packer"
+)
+
+type StepDeleteOSDisk struct {
```

Comment:
The storage account is not associated with the temporary resource group, so wiping out the resource group does not delete the VHD.  If the disk were inside the temporary resource group we would have to copy the VHD to a permanent storage account later.

I did not see an option in the delete VM API to delete the VHD, but I could have easily missed it.